src/vmspawn/vmspawn.c

```diff
@@ -601,15 +615,86 @@ static int find_initrd(char **ret_initrd) {
         return -ENOENT;
 }
 
+static int start_virtiofsd(sd_bus *bus, const char *scope, const char *directory, char **ret_state_tempdir, char **ret_sock_name) {
+        _cleanup_(rm_rf_physical_and_freep) char *state = NULL;
+        _cleanup_strv_free_ char **cmdline = NULL, **cleanup = NULL;
+        _cleanup_free_ char *virtiofsd = NULL, *sock_path = NULL, *sock_name = NULL, *rm_path = NULL, *unit_name = NULL;
+        int r;
+
+        assert(bus);
+        assert(scope);
+        assert(directory);
+        assert(ret_state_tempdir);
+        assert(ret_sock_name);
+
+        r = find_executable("virtiofsd", &virtiofsd);
+        if (r < 0 && r != -ENOENT)
+                return log_error_errno(r, "Error while searching for virtiofsd: %m");
+
+        if (!virtiofsd) {
+                FOREACH_STRING(dir, "/usr/libexec/virtiofsd", "/usr/libexec/virtiofsd") {
```

Comment:
hmm, which distro puts the binary there?

Reply:
the two paths are exacty the same?

Reply:
I'm not sure, this was behaviour inherited from mkosi: https://github.com/systemd/mkosi/blob/a4444f01d0e29e380c576cf218f3ef3dc66b8409/mkosi/qemu.py#L302-L312

Reply:
And on the second point this is a typo and the second one should be `/usr/lib/virtiofsd`

Reply:
@poettering Fedora puts it there

Reply:
@DaanDeMeyer is the version fedora puts there the Rust rewrite?

On my Ubuntu install it is still on the pre rust-rewrite version and its located at `/usr/lib/qemu/virtiofsd`.
As in should I include `/usr/lib/qemu/virtiofsd` as well as a fallback path.

The issue is that the arguments aren't compatible and it requires root so finding the old version wouldn't be desirable unless there is potential for the rust-rewrite version to appear there in future.

Reply:
@sam-leonard-ct We should only support the rust rewrite

Reply:
Fixed in eae47ee9d9.